armasec/armasec.py

```diff
@@ -44,16 +44,39 @@ def __init__(
         self.debug_exceptions = debug_exceptions
 
     @lru_cache(maxsize=128)
-    def lockdown(self, *scopes: str) -> TokenSecurity:
+    def lockdown(
+        self,
+        *scopes: str,
+        permission_mode: PermissionMode = PermissionMode.ALL,
+    ) -> TokenSecurity:
         """
         Initialize an instance of TokenSecurity to lockdown a route. Uses memoization to minimize
-        the number of TokenSecurity instances initialized.
+        the number of TokenSecurity instances initialized. Applies supplied permission_mode when
+        checking token permssions against TokenSecurity scopes.
         """
         return TokenSecurity(
             self.domain,
             audience=self.audience,
             algorithm=self.algorithm,
             scopes=scopes,
+            permission_mode=permission_mode,
             debug_logger=self.debug_logger,
             debug_exceptions=self.debug_exceptions,
         )
+
+    def lockdown_all(self, *scopes: str) -> TokenSecurity:
+        """
+        Initialize an instance of TokenSecurity to lockdown a route. Uses memoization to minimize
```

Comment:
`Uses memoization` - it does?

Reply:
Yes, the base `lockdown()` method uses the `lru_cache` from `functools`.